### ISSUE TITLE
convert on fp functions

### ISSUE THREAD

**Issue by olsonpm** [lodash/lodash#2318]
I'm a little confused by [convert](https://github.com/lodash/lodash/wiki/FP-Guide#convert)

It is attached to each fp function, but based on its current functionality, I would expect it to instead exist on each vanilla function.

To give some context, I got here from #2275 and expected the mutable assign to work as fp.assign does, with the vanilla argument order.

```js
const mutableAssign = require('lodash/fp/assign').convert({ immutable: false });

const a = { name: 'a' }
  , b = { name: 'b' };

mutableAssign(a, b);
console.log('a: ' + JSON.stringify(a, null, 2));
console.log('b: ' + JSON.stringify(b, null, 2));

// expected output
// a: {
//   "name": "b"
// }
// b: {
//   "name": "b"
// }

// actual
// a: {
//   "name": "a"
// }
// b: {
//   "name": "a"
// }
```

In order to get the expected results, I needed to specifify `{ rearg: false }`, meaning rearg is performed on the already converted fp function.  Is this by design?  As-is, it makes more sense to:
```js
const mutableAssign = require('lodash/assign').convert({ immutable: false });
```

Except convert obviously isn't defined on vanilla assign.

Edit: Got the expected/actual backwards

**Comment by jdalton** [comment]
The conversion has no added cost.
By that I mean it produces a freshly wrapped method (so no nesting of conversions).

The conversion process is an fp thing only at the moment (it's part of its module generation).

**Comment by olsonpm** [comment]
Alright - I'll just read it differently then.  I was just pointing out that `<fp module>.convert` is not converting the fp module, but the vanilla one.  I see where you wouldn't want to add to the vanilla api, and how placing `convert` onto the fp module is convenient, but semantically speaking if the convert function is being performed on the vanilla module, then it should live there.

**Comment by jdalton** [comment]
The fp modules are converted forms of the vanilla ones. While there's no nested conversions each conversion builds off of the options of the previous.
```js
import { _remove } from 'lodash/fp';
const remove = _.remove.convert({ 'cap': false }).convert{ 'rearg': false });
```

**Comment by olsonpm** [comment]
Thanks for the clarification.  Had to take a look at the lodash code to understand my problem better, and it seems the issue is that all methods are either converted with default values or _falseOptions, where _falseOptions are passed to vanilla unary methods.

So even though `fp.assign` does not have rearranged arguments, it's not a result of `rearg` being set to false during the conversion, it's just that lodash's build process was configured to skip the rearg during conversion (found in _mapping -> skipRearg).

Regardless, I hadn't realized convert was its own module until figuring this out, which resolves my problem because I can convert vanilla methods as it plainly shows in the documentation.

**Comment by jdalton** [comment]
Hmm reviewing your issue again I can't repro it. I get the expected result with v4.12.0:
```js
// expected output
// a: {
//   "name": "b"
// }
// b: {
//   "name": "b"
// }
```

**Comment by olsonpm** [comment]
ugh - I was on lodash@4.7 (npm).  Apparently this was fixed in 4.8.  I will make sure to check this in the future before writing an issue.
